Drop corepack: use the installed pnpm locally and the latest pnpm in CI

## Summary

Stop using corepack to manage pnpm. Locally we should just use whatever pnpm is already installed on the machine, and in CI we should install/use the latest pnpm rather than a pinned version resurrected by corepack.

Today the pnpm version is pinned via the `packageManager` field in [`ui/menu-website/package.json`](ui/menu-website/package.json), corepack is what Aspire's `WithPnpm()` shells out to when the AppHost starts the stack, and CI explicitly runs `corepack enable pnpm`. That pin is a recurring maintenance cost (a Dependabot-adjacent bump nobody actually wants to review) and it produces the stale-`node_modules` purge failure we already had to work around.

## Current state

- `ui/menu-website/package.json` — `"packageManager": "pnpm@11.21.0+sha512...."` (line 87)
- `.github/workflows/main.yml` — `- name: Enable pnpm with corepack` / `run: corepack enable pnpm` (lines 215–216) in the `frontend` job
- `ui/menu-website/pnpm-workspace.yaml` — `confirmModulesPurge: false` plus a comment block explaining it exists specifically because a **corepack pin bump** leaves a stale module store
- `backend/Menu.AppHost/Program.cs:27-28` — `AddJavaScriptApp("menu-ui", …).WithPnpm()`, which resolves `pnpm` off `PATH`
- `AGENTS.md` — documents the corepack pin and the `ERR_PNPM_ABORTED_REMOVE_MODULES_DIR_NO_TTY` workaround

## Proposed change

**Local**

- Remove the `packageManager` field from `ui/menu-website/package.json`, so `pnpm` resolves to whatever the developer has installed globally (and so `WithPnpm()` uses that same binary).
- Optionally add a `pnpm` entry to the existing `engines` block as a floor (e.g. `">=11"`) so a genuinely ancient pnpm still fails fast — a floor, not a pin.
- Keep `confirmModulesPurge: false` in `pnpm-workspace.yaml`, but update the comment: the trigger is no longer a corepack pin bump, it's any pnpm major-version change between the developer's global pnpm and the store on disk. The non-TTY failure mode under Aspire is unchanged.

**CI** (`.github/workflows/main.yml`, `frontend` job)

- Replace `corepack enable pnpm` with an explicit install of the latest pnpm — either `pnpm/action-setup@v4` with `version: latest`, or a plain `npm install -g pnpm@latest`. Use whichever plays nicest with the existing `pnpm store path` cache step that follows.
- Keep the rest of the job (store cache, `pnpm install --frozen-lockfile`, generate/lint/build/test) unchanged.
- Confirm `codeql.yml` doesn't need the same treatment (it currently has no pnpm setup).

**Docs**

- Update `AGENTS.md` (the pnpm/corepack paragraph in the frontend section) and any other doc referencing the corepack pin, so agents and contributors don't reintroduce it.

## Risks / things to check

- **Lockfile drift.** The pin is the only thing currently guaranteeing everyone writes the same `pnpm-lock.yaml` format. With CI on `latest` and developers on whatever they have installed, a pnpm major that bumps `lockfileVersion` will show up as a large unrelated lockfile diff, and `--frozen-lockfile` in CI will fail if a developer's older pnpm wrote an older format. This is the main trade-off to accept knowingly.
- **Corepack still enabled globally.** If a developer has `COREPACK_ENABLE_STRICT` set, removing `packageManager` should be a non-issue, but worth sanity-checking on a clean machine.
- **Aspire path.** Verify `aspire start` / `dotnet run --project Menu.AppHost` still starts the UI once corepack is out of the picture — `WithPnpm()` needs a `pnpm` on `PATH`.

## Acceptance criteria

- [ ] No `packageManager` field remains in `ui/menu-website/package.json`
- [ ] No `corepack` invocation remains in `.github/workflows/`
- [ ] CI frontend job installs and uses the latest pnpm, and the store cache still hits
- [ ] `pnpm install`, `pnpm build`, `pnpm lint`, `pnpm test` all pass in CI
- [ ] AppHost (`aspire start`) still launches the UI locally
- [ ] `AGENTS.md` and `pnpm-workspace.yaml` comments reflect the new setup, with no stale corepack references

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -212,8 +212,8 @@ jobs:
         with:
           node-version: 22.x
 
-      - name: Enable pnpm with corepack
-        run: corepack enable pnpm
+      - name: Install latest pnpm
+        run: npm install -g pnpm@latest
 
       - name: Get pnpm store directory
         id: pnpm-cache

--- a/AGENTS.md
+++ b/AGENTS.md
@@ -170,7 +170,7 @@ pnpm storybook            # Storybook dev server (port 6006)
 
 Always run `pnpm test:storybook` after making any change under `ui/menu-website/src/` — it catches regressions in components exercised by existing stories, not just changes to story files themselves.
 
-pnpm's version is pinned via corepack (`packageManager` in `package.json`), which is also what Aspire's `WithPnpm()` shells out to when the AppHost starts the full stack (via `aspire start`, the agent-preferred path, or `dotnet run --project Menu.AppHost`). `pnpm-workspace.yaml` sets `confirmModulesPurge: false` so a pinned-version bump self-heals a stale `node_modules` automatically; if you ever see `ERR_PNPM_ABORTED_REMOVE_MODULES_DIR_NO_TTY` anyway, just run `pnpm install` once by hand from `ui/menu-website/`.
+pnpm's version is not pinned — whatever `pnpm` is on your `PATH` is what runs, including when Aspire's `WithPnpm()` shells out to it to start the full stack (via `aspire start`, the agent-preferred path, or `dotnet run --project Menu.AppHost`). `package.json` declares a `>= 10` floor under `engines`, and CI installs the latest pnpm; do not reintroduce a `packageManager` pin or corepack. `pnpm-workspace.yaml` sets `confirmModulesPurge: false` so upgrading your global pnpm across a major self-heals a stale `node_modules` automatically; if you ever see `ERR_PNPM_ABORTED_REMOVE_MODULES_DIR_NO_TTY` anyway, just run `pnpm install` once by hand from `ui/menu-website/`.
 
 ### Component Test Coverage
 

--- a/AGENTS.md
+++ b/AGENTS.md
@@ -170,7 +170,7 @@ pnpm storybook            # Storybook dev server (port 6006)
 
 Always run `pnpm test:storybook` after making any change under `ui/menu-website/src/` — it catches regressions in components exercised by existing stories, not just changes to story files themselves.
 
-pnpm's version is not pinned — whatever `pnpm` is on your `PATH` is what runs, including when Aspire's `WithPnpm()` shells out to it to start the full stack (via `aspire start`, the agent-preferred path, or `dotnet run --project Menu.AppHost`). `package.json` declares a `>= 10` floor under `engines`, and CI installs the latest pnpm; do not reintroduce a `packageManager` pin or corepack. `pnpm-workspace.yaml` sets `confirmModulesPurge: false` so upgrading your global pnpm across a major self-heals a stale `node_modules` automatically; if you ever see `ERR_PNPM_ABORTED_REMOVE_MODULES_DIR_NO_TTY` anyway, just run `pnpm install` once by hand from `ui/menu-website/`.
+pnpm's version is not pinned — whatever `pnpm` is on your `PATH` is what runs, including when Aspire's `WithPnpm()` shells out to it to start the full stack (via `aspire start`, the agent-preferred path, or `dotnet run --project Menu.AppHost`). `package.json` declares a `>= 10` floor under `engines`, and CI installs the latest pnpm; do not reintroduce a `packageManager` pin or corepack. `pnpm-workspace.yaml` sets `confirmModulesPurge: false` so upgrading your global pnpm across a major version self-heals a stale `node_modules` directory automatically; if you ever see `ERR_PNPM_ABORTED_REMOVE_MODULES_DIR_NO_TTY` anyway, just run `pnpm install` once by hand from `ui/menu-website/`.
 
 ### Component Test Coverage
 

--- a/ui/menu-website/package.json
+++ b/ui/menu-website/package.json
@@ -82,9 +82,9 @@
   "engines": {
     "node": "^22.22.2 || ^24.15.0 || >=26.0.0",
     "npm": ">= 6.13.4",
+    "pnpm": ">= 10",
     "yarn": ">= 1.21.1"
   },
-  "packageManager": "pnpm@11.21.0+sha512.521705bce689924eac72f5a3587122f362689ef6571e55ba80076fd637c11132ecffada26fad4ea79c485bfddbfd3d5a2a5b05805a77e893de71ec8a6cca3bb1",
   "msw": {
     "workerDirectory": [
       "public"

--- a/ui/menu-website/pnpm-workspace.yaml
+++ b/ui/menu-website/pnpm-workspace.yaml
@@ -1,8 +1,8 @@
 # Aspire's WithPnpm() spawns pnpm non-interactively with no TTY. Without this,
-# a pnpm major-version bump (see the corepack packageManager pin in package.json)
-# leaves a stale node_modules store from the old version, and pnpm aborts waiting
-# for a purge confirmation prompt it can never receive
-# (ERR_PNPM_ABORTED_REMOVE_MODULES_DIR_NO_TTY).
+# any pnpm major-version change between the pnpm on your PATH and the store on
+# disk (e.g. after upgrading your global pnpm) leaves a stale node_modules from
+# the old version, and pnpm aborts waiting for a purge confirmation prompt it
+# can never receive (ERR_PNPM_ABORTED_REMOVE_MODULES_DIR_NO_TTY).
 confirmModulesPurge: false
 allowBuilds:
   '@parcel/watcher': true

--- a/ui/menu-website/pnpm-workspace.yaml
+++ b/ui/menu-website/pnpm-workspace.yaml
@@ -1,8 +1,9 @@
 # Aspire's WithPnpm() spawns pnpm non-interactively with no TTY. Without this,
-# any pnpm major-version change between the pnpm on your PATH and the store on
-# disk (e.g. after upgrading your global pnpm) leaves a stale node_modules from
-# the old version, and pnpm aborts waiting for a purge confirmation prompt it
-# can never receive (ERR_PNPM_ABORTED_REMOVE_MODULES_DIR_NO_TTY).
+# a pnpm major-version mismatch — the pnpm on your PATH differing from the one
+# that wrote node_modules, e.g. after upgrading your global pnpm — leaves behind
+# a node_modules directory pnpm considers stale. pnpm then aborts waiting for a
+# purge confirmation prompt it can never receive
+# (ERR_PNPM_ABORTED_REMOVE_MODULES_DIR_NO_TTY).
 confirmModulesPurge: false
 allowBuilds:
   '@parcel/watcher': true

